substrate/client/consensus/beefy/src/worker.rs

```
@@ -747,23 +753,38 @@ where
 			);
 			return Ok(())
 		};
-
+		
+		let payload = Payload::from_single_entry(known_payloads::ETF_SIGNATURE, Vec::new());
 		let commitment = Commitment { payload, block_number: target_number, validator_set_id };
 		let encoded_commitment = commitment.encode();
 
-		let signature = match self.key_store.sign(&authority_id, &encoded_commitment) {
-			Ok(sig) => sig,
-			Err(err) => {
-				warn!(target: LOG_TARGET, "🥩 Error signing commitment: {:?}", err);
+		let signature = match self.etf_extract(
+			target_hash, 
+			authority_id.clone(), 
+			&encoded_commitment,
+		) {
+			Some(sig) => sig,
+			None => {
+				warn!(target: LOG_TARGET, "🥩 Error calculating ETF signature");
 				return Ok(())
 			},
 		};
+		
+		let etf_authority_id = self.runtime.runtime_api()
+			.read_commitment(target_hash, authority_id.clone())
+			.ok()
+			.unwrap()
+			.expect("todo");
```

Can we return an Error here?